nengo/tests/test_learning_rules.py

```diff
@@ -466,3 +467,118 @@ def test_frozen():
 
     with pytest.raises((ValueError, RuntimeError)):
         a.learning_rate = 1e-1
+
+
+@pytest.mark.parametrize('modifies', ["decoders", "weights"])
+def test_generic_rule(Simulator, nl_nodirect, seed, modifies):
+    if nl_nodirect == nengo.RectifiedLinear and modifies == "weights":
+        # this is a sloppy way of avoiding this one case without having
+        # to make a big custom parameterization
+        # TODO: why doesn't it work for this case?
+        return
+
+    n = 100
+    input_vector = np.asarray([0.5, -0.5])
+    dt = 0.001
+    reg = 0.1
+
+    m = nengo.Network(seed=seed)
+    with m:
+        m.config[nengo.Ensemble].neuron_type = nl_nodirect()
+        u = nengo.Node(output=input_vector)
+        a = nengo.Ensemble(n, dimensions=2)
+        b = nengo.Ensemble(n, dimensions=2)
+
+        nengo.Connection(u, a)
+
+        if modifies == "decoders":
+            def regularized_pes(target, data):
+                return -dt / n * np.outer(data[:2], data[2:]) - target * reg
+        else:
+            def regularized_pes(target, data, params):
+                encoders = params[b].encoders
+                gains = params[b].gain
+                return (-dt / np.sqrt(n) *
+                        np.outer(np.dot(encoders, data[:2]) * gains,
+                                 data[2:]) - target * reg)
+
+        rule = GenericRule(regularized_pes, learning_rate=1e-3,
+                           size_in=2 + n, modifies=modifies,
+                           pass_model_params=modifies == "weights")
+
+        conn = nengo.Connection(
+            a if modifies == "decoders" else a.neurons,
+            b if modifies == "decoders" else b.neurons,
+            transform=1 if modifies == "decoders" else np.eye(n),
+            learning_rule_type=rule)
+        nengo.Connection(b, conn.learning_rule[:2], transform=1)
+        nengo.Connection(u, conn.learning_rule[:2], transform=1)
+        nengo.Connection(a.neurons, conn.learning_rule[2:], synapse=0.005)
+
+        b_p = nengo.Probe(b, synapse=0.1)
+        weights_p = nengo.Probe(conn, attr="weights")
+
+    sim = Simulator(m, dt=dt)
+    sim.run(1.)
+
+    tmask = sim.trange() > .9
+    assert (np.sum(np.abs(sim.data[weights_p][:10])) >
+            np.sum(np.abs(sim.data[weights_p][-10:])))
```

Comment:
Because our learning rule is applying an l2 regularization term (see the `regularized_pes` function above)